### ISSUE TITLE
[Homebridge UI] Restore backup failed: zlib: unexpected end of file when trying to restore a backup on new HB instance on new machine.

### ISSUE THREAD

**Issue by buddyrich33** [oznu/homebridge-config-ui-x#1856]
### Current Situation

Trying to move my homebridge instance from one Mac to a new one and I am getting this error.
[11/26/2023, 10:46:17 AM] [Homebridge UI] Restore backup failed: zlib: unexpected end of file

The homebridge I am trying to back up is a Mac Mini 2018 (so intel CPU) running latest MacOS Sonoma.
homebridge is 1.7.0 plugins and Node version are up to date (node version 20.10.0)...  I can make and download my backup file fine.  I stop my hb instance via hb service command,.

The new mac I am trying to restore to is an M2 Max Mac Studio, running 1.7 homebridge and same 20.10.0 node version. No plugins yet..  I believe the restore will install them.

The backup file itself is 1.7 MBs and unzips with no errors and I can see the config files.. so it doesn't seem to be a corrupt backup.

### Logs

```text
Even with debug logging on this is the only error I get in the logs:

[11/26/2023, 10:46:17 AM] [Homebridge UI] Restore backup failed: zlib: unexpected end of file
```

### Homebridge UI Version

4.53.0

### Homebridge Version

1.7.0

### Node.js Version

20.10.0

### Operating System

macOS

### Environment Info

- [ ] Using Docker?
- [ ] Using Hyper-V?
- [X] Using hb-service?

### Raspberry Pi Model

None

**Comment by bwp91** [comment]
Hi @buddyrich33 is your version of Homebridge UI up to date? The latest (at time of my comment) is v4.53.0, do you get this error message about restoring on this version of UI?

**Comment by buddyrich33** [comment]
Yes.  It was 4.53.0...  Though I solved my issue, though I don't know which file was the culprit.  

I thought perhaps my backup file was too large, at around 1.6MB...  

I unziped the backup file and saw the expected configuration jsons, etc...  in the storage folder I noticed two plugins in particular also had gziped logs...  hue plugin, had maybe a 100KB or so so log, but the hubitat maker-api had multipart logs, with the active one almost 7MB in size.  There was also a strange zero-byte file called <machine name>_persist.json...  Anyway I deleted the logs and the zerobyte file, re gziped them up and was able to use the much smaller backup file to restore successfully.

**Comment by sapireli** [comment]
This is a recurring issue with the latest version. The error is the same always: "Restore backup failed: zlib: unexpected end of file" 

Testing was done on a raspi running debian bookworm image of dietpi. 

The backup comes in at 6.3mb compressed.

Upon closer inspection: there is a large hidden ".npm" folder inside the backup. This folder is a relatively large sub-folder -> _cacache/content-v2/sha512/ -> which contains over 100 two letter folders. Removing this folder and gzipping allows the backup file to be successfuly uploaded but then the terminal window reports the following error: "Error: Uploaded file is not a valid Homebridge Backup Archive."
"

**Comment by NorthernMan54** [comment]
@sapireli - These are the conditions that trigger that message - https://github.com/homebridge/homebridge-config-ui-x/blob/f63405f68a55416be3f9bb3ee4d47227b78d691c/src/modules/backup/backup.service.ts#L407

Your rebuilt zip file must be missing one of them

PS - Is it possible to share the broken backup file to me, so we can look at remediating the issue ?  I'm on the Homebridge discord server so we could leverage that.

**Comment by sapireli** [comment]
> @sapireli - These are the conditions that trigger that message -
> 
> https://github.com/homebridge/homebridge-config-ui-x/blob/f63405f68a55416be3f9bb3ee4d47227b78d691c/src/modules/backup/backup.service.ts#L407
> 
> Your rebuilt zip file must be missing one of them
> 
> PS - Is it possible to share the broken backup file to me, so we can look at remediating the issue ? I'm on the Homebridge discord server so we could leverage that.

Thank you so much for pointing me there, i submitted the fix in  #1948 its working well now.

**Comment by StefanGebhardt** [comment]
Bug is still there in 4.54.2-beta.58
Updated to this version on the source installation, made a backup, tried to restore in the new installed installation (also 4.54.2-beta.58). Still the same error...

**Comment by bwp91** [comment]
Hi @StefanGebhardt 

assuming you are still on the beta version, can you generate and download a new backup file, download and locate it in your finder/windows explorer and see how large the file is?

**Comment by StefanGebhardt** [comment]
It has 15 MB (14.841.218 bytes)

**Comment by bwp91** [comment]
Are you able to extract it on your local file system?

**Comment by StefanGebhardt** [comment]
Yes, with no problems on Mac OS. Need any information about the files inside?

**Comment by bwp91** [comment]
Just knowing what is generally taking up the most space would be useful.

I’m unsure of what a typical backup would be in terms of size, but 15MB seems a lot for just some text files.

Perhaps there are more directories we need to ignore. and perhaps this error is showing when the file is simply too large 🧐

**Comment by StefanGebhardt** [comment]
These are the files & directories after extracting:

![Untitled](https://github.com/homebridge/homebridge-config-ui-x/assets/2803552/a6ab5f7a-79d6-482a-b75c-7bce4bc1fe97)

**Comment by StefanGebhardt** [comment]
I'll tried a lot of deleting and compacting again. Especially the big *persist.json-Files. Nothing works... :-(

**Comment by bwp91** [comment]
@StefanGebhardt are you able to provide another screenshot with those top folders expanded to see which files in them are taking up the most space?

I've done a bit of testing locally, and it seems that whilst uncompressing, if the new folder gets over ~1.2mb then this error is thrown.

**Comment by bwp91** [comment]
Hold that above thought above

@StefanGebhardt @buddyrich33 @sapireli 

Are you able to test a restore process on the beta version of UI? At least v4.54.2-beta.78 ?

**Comment by sapireli** [comment]
i just tested backup and restore on the 4.54.2-beta.79, and it is working well. 

@StefanGebhardt Please expand all the folders, and show hidden files as well. Something seems off.

**Comment by sapireli** [comment]
FWIW my backup of an extensive homebridge config clocks in at a heft weight of 125 KBs on the beta.

**Comment by sapireli** [comment]
@StefanGebhardt also try running "npm cache clean --force" on your homebridge server, then run the backup again see if the backup size gets smaller...

**Comment by bwp91** [comment]
@StefanGebhardt no need do any of this

**Comment by bwp91** [comment]
sorry @sapireli didnt mean to overrule you there but there indeed was a an issue with the UI, so any backups should be restore-able now

**Comment by StefanGebhardt** [comment]
@bwp91 Updated to Beta.79 and it worked without any problems. Thank you so much!

**Comment by Goldsucher** [comment]
I still have this error with latest version of homebridge-ui.
My backup file is 44.4 MB. There are many .zigbee files which blow up the file (I use homebridge-deconz)
Do I need these .zigbee files? I tried to remove them and compress the folder with Keka (Mac App) but then I get the error "Error: Uploaded file is not a valid Homebridge Backup Archive."

Could someone help me with this?

EDIT: Okay I deletted the files, made the backup und after the restore everything still works 👍
